docs/algorithm-registry.md

```diff
@@ -0,0 +1,17 @@
+# Algorithm Registry
+
+This file is designed to act as a source of truth regarding what signing
+algorithms are permitted across the Sigstore ecosystem. Any changes to this file
+**must** be reflected in the `SupportedAlgorithm` enumeration in
+[sigstore_common.proto](../protos/sigstore_common.proto).
+
+Refer to the [Sigstore: Configurable Crypto Algorithms](https://docs.google.com/document/d/18vTKFvTQdRt3OGz6Qd1xf04o-hugRYSup-1EAOWn7MQ/)
+specification for the design rationale for this registry.
+
+| Algorithm | Name | Usage |
+| --- | --- | --- |
+| ECDSA | ecdsa-sha2-256-nistp256 | sign/verify |
+|| ecdsa-sha2-256-nistp521 | verify only |
+|| ecdsa-sha2-384-nistp384 | sign/verify |
+| EdDSA | ed25519 | sign/verify |
+|| ed25519-ph | sign/verify |
```

Comment:
Possibly not important, but we may want to omit `ed25519` for now -- `hashedrekord` will use the `ed25519-ph` variant, and I'm not sure we have an immediate use for the non-prehashed "normal" variant.

(But I might be missing a use case.)

CC @haydentherapper

Reply:
It seems like rekor accepts ED25519 keys with SHA256 for example when used with attest-blob/verify-blob-attestation in cosign.

Reply:
Re: https://github.com/sigstore/protobuf-specs/pull/194/files#r1453680630 as well: we'll need bare `ed25519` without the prehashing for `dsse` entries.

Reply:
Cool, the takeaway here is:
- Keep Ed25519 here.
- Make a note that ed25519-ph is only for hashedrekord.

Reply:
[1243b19](https://github.com/sigstore/protobuf-specs/pull/194/commits/1243b19b6394046598dfc7374f3084fe98b3f222) and [c06b5d1](https://github.com/sigstore/protobuf-specs/pull/194/commits/c06b5d1e850c96976ba6b1956e88fda3d3341b1d)